libs/geometry/src/TangentSpaceMeshInternal.h

```diff
@@ -22,11 +22,42 @@
 #include <math/mat3.h>
 #include <math/norm.h>
 
+#include <vector>
+
 namespace filament::geometry {
 
 using namespace filament::math;
 using Algorithm = TangentSpaceMesh::Algorithm;
 
+template<typename T>
+class InternalArray {
+public:
+    void borrow(const T* ptr) noexcept {
+        mBorrowed = ptr;
+    }
+
+    T* allocate(size_t size) noexcept {
```

Comment:
Should it be an error to allocate() a borrowed vector? assert?

Reply:
done